### DISCUSSION TITLE
Unable to send emails via postal.

### DISCUSSION THREAD

**Discussion by jonny-portridge** [postalserver/postal#2419] [Help with using Postal]
Hi, 

I have setup a postal server, everything appears to be set up correctly, however I can't send emails via postal. I have created a company, added and verified a domain in the web portal, but I can't send email. 

I have tried connecting via telnet, and I can authenticate with the credentials provided in postal. But when I try to send a test email, I get the error from sender name not valid. But the domain matches the domain added in postal. 

I've also set up port 587 to port 25 via iptables in case it was the web host blocking port 25, but neither work. 

MacBook-Pro-2:~ jonny$ nc mail.portridge.uk 587
220 mail.portridge.uk ESMTP Postal/LXLBGU
ehlo
250-My capabilities are
250 AUTH CRAM-MD5 PLAIN LOGIN
auth login
334 VXNlcm5hbWU6
base64ecoded username
334 UGFzc3dvcmQ6
base64encoded password
235 Granted for postaluser/postal-server
mail from: test@pgdev.uk
250 OK
rcpt to: admin@portridge.uk
250 OK
data
354 Go ahead
subject: testing postal
Hello!
.
530 From/Sender name is not valid.

![Screenshot 2023-04-20 at 12 20 24](https://user-images.githubusercontent.com/103172850/233351362-2a292407-3e50-4451-990b-9b01b8c6ea73.png)

Any ideas why I can't get this to work? 

Thanks

**Comment by willpower232** [comment]
In this case, you need to add a "from" header in the message (i.e. before/after your subject line). This would usually be done for you if you were using SMTP-capable software rather than telnet directly.

**Reply by jonny-portridge** [reply]
Hi @willpower232, I was just trying to do this as a test to see if it would work, because it doesn't work from Wordpress. Which is where I need to send the emails from.  We use the WP Forms SMTP plugin. https://wordpress.org/plugins/wp-mail-smtp But I have also installed a few different plugins and it will not send from any of them via postal. 

![Screenshot 2023-04-20 at 14 33 06](https://user-images.githubusercontent.com/103172850/233382661-82fe402d-5446-491b-adad-1c7677765391.png)

![Screenshot 2023-04-20 at 14 35 13](https://user-images.githubusercontent.com/103172850/233383032-65362ee2-684e-4ea5-aad1-cc0ffdcc20f8.png)

**Reply by jonny-portridge** [reply]
It is connecting successfully to the postal server, because all my test emails are queued, in the messages tab. But they all have pending status and none have been sent. 

I have confirmed with the server host that port 25 outbound is open with them. It was blocked and I had to open a support ticket to open it up.

**Reply by willpower232** [reply]
The messages are queued but not being processed? check `postal status` to be sure you're running every part of Postal.

**Reply by jonny-portridge** [reply]
Hi Will, yes they are all in outgoing messages as pending in the web console. 

<img width="1133" alt="Screenshot 2023-04-21 at 12 44 02" src="https://user-images.githubusercontent.com/103172850/233627526-b58d53ce-f794-4714-b8f5-61446dfe48d9.png">

here is the status

root@vultr:~# postal status
NAME                IMAGE                               COMMAND                  SERVICE             CREATED             STATUS              PORTS
postal-cron-1       ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   cron                3 days ago          Up 3 days           
postal-smtp-1       ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   smtp                3 days ago          Up 3 days           
postal-web-1        ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   web                 3 days ago          Up 3 days

**Reply by jonny-portridge** [reply]
I noticed that there should be postal-worker-1 and postal-requeuer-1 there also, so I stopped initilized and started again and now everything appears to be running. But still no mail. 

I have also just connect via telnet again and added the from header as you suggested and that retuned no errors in the console, but the email is just pending in the outgoing queue again.

**Reply by willpower232** [reply]
You seem to be missing the workers? Those are what sends out messages and does all of the queued tasks.

Have a look at the default docker-compose.yml for what is required these days https://github.com/postalserver/install/blob/main/templates/docker-compose.yml

**Reply by willpower232** [reply]
Oops sorry, if your workers had crashed then the database could have the messages assigned to workers which don't exist any more so won't get sent.

Have a look at https://github.com/postalserver/postal/issues/287#issuecomment-367293435

**Reply by jonny-portridge** [reply]
Sorry, I'm not sure what that means, there is no mention of editing that file in the setup guide and I don't know where it is. I have flattened the server and reinstalled 3 times now and it is the same every time. 

I'm using this guide https://computingforgeeks.com/install-and-configure-postal-mail-server-on-ubuntu/ 

I run the set up everything appears to be working, I can log into the web console and create a server, add domain and credentials, then I try to send an email and they just sit in the queue as pending. 

The output below is postal status shows everything running, install caddy, connect to the web interface, set up server domain, try to send an email from wordpress, doesn't send, check status again. 

Is there something else that I should be doing/checking? I tried to use the official guide installing mariadb and rabbitmq in docker container, but got a mysql error no access for root@localhost, so followed that quide instead, installing them on the server which worked. 

root@vultr:~# postal start
[+] Running 5/5
 ✔ Container postal-worker-1    Started                                                                                                  0.4s 
 ✔ Container postal-requeuer-1  Started                                                                                                  0.3s 
 ✔ Container postal-cron-1      Started                                                                                                  0.5s 
 ✔ Container postal-web-1       Started                                                                                                  0.5s 
 ✔ Container postal-smtp-1      Started                                                                                                  0.4s 
root@vultr:~# postal status
NAME                IMAGE                               COMMAND                  SERVICE             CREATED             STATUS              PORTS
postal-cron-1       ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   cron                5 seconds ago       Up 4 seconds        
postal-requeuer-1   ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   requeuer            5 seconds ago       Up 4 seconds        
postal-smtp-1       ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   smtp                5 seconds ago       Up 4 seconds        
postal-web-1        ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   web                 5 seconds ago       Up 4 seconds        
postal-worker-1     ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   worker              5 seconds ago       Up 4 seconds        
root@vultr:~# docker run -d \
   --name postal-caddy \
   --restart always \
   --network host \
   -v /opt/postal/config/Caddyfile:/etc/caddy/Caddyfile \
   -v /opt/postal/caddy-data:/data \
   caddy
Unable to find image 'caddy:latest' locally
latest: Pulling from library/caddy
91d30c5bc195: Pull complete 
7f137c1fd65a: Pull complete 
123731571dfc: Pull complete 
9ab4cbb8b7b7: Pull complete 
Digest: sha256:cb9a70e2fe64b1f0a871628e2fa71efcfb3f632dacc0a210ca8d484445cca70a
Status: Downloaded newer image for caddy:latest
8214a3c11d47759b5c1a043e9b78b71da38eb4b168b194948d45506787b7737d
root@vultr:~# postal statusNAME                IMAGE                               COMMAND                  SERVICE             CREATED             STATUS              PORTS
postal-cron-1       ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   cron                4 minutes ago       Up 4 minutes        
postal-smtp-1       ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   smtp                4 minutes ago       Up 4 minutes        
postal-web-1        ghcr.io/postalserver/postal:2.1.2   "/docker-entrypoint.…"   web                 4 minutes ago       Up 4 minutes        
root@vultr:~#

**Reply by willpower232** [reply]
If Postal is running then that is all fine, messages are queued in rabbitmq but then "locked" by a row in the database so they aren't sent multiple times.

If your worker is having a bad time then that means messages will sit in the database "locked" and not be sent so you need to remove those "locks" as explained in https://github.com/postalserver/postal/issues/287#issuecomment-367293435

**Reply by jonny-portridge** [reply]
Okay I tried that, but it doesn't change anything, and the messages still sit in the queue as pending.  Think it might be time to call it quits and find a different solution. 

MariaDB [(none)]> use postal
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
MariaDB [postal]> select * from queued_messages;
+----+-----------+------------+--------------+-----------+-----------+-------------+----------------------------+----------------------------+---------------+----------+----------+--------+-----------------------+
| id | server_id | message_id | domain       | locked_by | locked_at | retry_after | created_at                 | updated_at                 | ip_address_id | attempts | route_id | manual | batch_key             |
+----+-----------+------------+--------------+-----------+-----------+-------------+----------------------------+----------------------------+---------------+----------+----------+--------+-----------------------+
|  1 |         1 |          1 | portridge.uk | NULL      | NULL      | NULL        | 2023-04-21 20:26:23.857307 | 2023-04-21 20:26:23.857307 |          NULL |        0 |     NULL |   NULL | outgoing-portridge.uk |
|  2 |         1 |          2 | portridge.uk | NULL      | NULL      | NULL        | 2023-04-22 09:47:30.349070 | 2023-04-22 09:47:30.349070 |          NULL |        0 |     NULL |   NULL | outgoing-portridge.uk |
+----+-----------+------------+--------------+-----------+-----------+-------------+----------------------------+----------------------------+---------------+----------+----------+--------+-----------------------+
2 rows in set (0.000 sec)

MariaDB [postal]> update queued_messages set locked_by = NULL, locked_at = NULL, retry_after = NULL; 
Query OK, 0 rows affected (0.000 sec)
Rows matched: 2  Changed: 0  Warnings: 0

**Reply by jonny-portridge** [reply]
It's okay, I managed to get it working!! There must have been an issue with the password I was using for the databases, when running the docker install for mariadb and rabbitmq. So then it caused the credentials to be invalid in the postal.yml and it failed to set up the database correctly. 

I tried the install again using a shorter password and the postal initialize completed successfully this time. 

Now that it is all set up on docker as per the guide, rather than installing the databases locally everything works as expected. 

Thanks for your help Will.

**Comment by uhlhosting** [comment]
@jonny-portridge hi, was wondering what plugin you use to send mails with postal in wp?

**Reply by jonny-portridge** [reply]
WP Mail Smtp, but any will work. 
https://wordpress.org/plugins/wp-mail-smtp/
